internal/storage/bucket_handle_test.go

```diff
@@ -365,6 +365,8 @@ func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsDefault() {
 }
 
 func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectExist() {
+	// Note: ContinuationToken doesn't work with fake storage and hence we can't
+	// add unit tests to test its functionality.
```

Comment:
why can't we add a test case to list 10 objects when bucket has 5 files + 5 folders + 5 files and how the data will be returned. when we set the max results to 10 , we should get 5 files + 5 folders, earlier it must be returning all.

Reply:
Thanks for this idea. I tried this and it worked. I will add a test case for this.

Reply:
With the new fix, this is not possible because the fake storage will always return all the objects and the new depends on the storage to return correct number of objects according to the value of maxResults.